lib/jsdom/living/events/MouseEvent-impl.js

```diff
@@ -16,31 +16,43 @@ class MouseEventImpl extends UIEventImpl {
   }
   get pageX() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._initialContainingBlockRelativeX;
     }
     const offset = wrapperForImpl(this.view)?.scrollX || 0;
     return offset + this.clientX;
   }
   get pageY() {
     if (this._dispatchFlag) {
-      return 0;
+      return this._initialContainingBlockRelativeY;
```

Comment:
This way, we're returning `clientY` adjusted to the `scrollY` value at the time when the event was created. Which could be a long time ago. If the viewport has scrolled in the meantime, the `pageY` value returning during dispatch won't reflect that.

That seems wrong and testing this in Chrome, I see different behavior:
1. After creating the event with `new MouseEvent`, I can inspect its `.pageY` field and it will always reflect the currrent scroll position. We are outside dispatch here. Your code implements this correctly.
2. If I scroll to another position and then dispatch the event, event listener (inside dispatch) will report `.pageY` value that reflects the current scroll position. Your code doesn't do that. It will use the scroll position from the time when the event was created. This value is even older than the one returned in step 1 when reading `.pageY` outside dispatch.
3. The `.pageY` value will probably be frozen during the entire dispatch. Setting `window.scrollY` inside a listener, or scrolling during the dispatch (is that even possible? event dispatch is completely synchronous) won't change the `.pageY` value.

Reply:
While this PR might seem wrong in terms of Chrome compatibility, I believe it's right in terms of following the spec. Given the browsers are all over the place and internally inconsistent (see below), wouldn't that be the best course?

Here's what my testing shows for `pageY` and `offsetY`... I got slightly different results that what you described, in particular around scrolling, though maybe I overlooked something? I should point out that you can synchronously scroll during dispatch (e.g. via `window.scroll(...)`), see the novel below and this note[^1] for more info 😄 

**Legend:**
* ✅ = implementation matches the spec and/or everybody agrees
* ❌ = doesn't match the spec but another implementation does
* 🤷 = spec is unclear and implementations disagree

Given:
```html
<div style="height:1000px; margin-top: 500px">
  <button>Go!</button>
</body>
<script>
  const evt = new MouseEvent("click", { view: window, clientY: 123 });
  const btn = document.querySelector("button");
  btn.addEventListener("click", (e) => {
    // window.scroll(0, 100); // optionally test browser scrolling during dispatch
    // window.scrollY = 100; // optionally test jsdom "scrolling" during dispatch
    console.log({pageY: e.pageY, offsetY: e.offsetY});
  });
</script>
```

1. If I scroll[^1] (outside of dispatch):
   * **`pageY`**
       * Chrome: reflects the event's `clientY` value but never considers `scrollY` (initial or current), i.e. `123` ❌
       * WebKit: reflects the event's `clientY` value but never considers `scrollY` (initial or current), i.e. `123` ❌
       * Gecko: reflects the event's `clientY` value plus the current `scrollY` value ✅ 
       * jsdom (prior to this PR): reflects the event's `clientY` value plus the current `scrollY` value ✅ 
       * jsdom (with this PR): reflects the event's `clientY` value plus the current `scrollY` value ✅ 
   * **`offsetY`**
       * Chrome: reflects the event's incorrect `pageY` value ❌
       * WebKit: reflects the event's incorrect `pageY` value ❌
       * Gecko: reflects the event's correct `pageY` value ✅ 
       * jsdom (prior to this PR): reflects the event's correct `pageY` value ✅ 
       * jsdom (with this PR): reflects the event's correct `pageY` value ✅ 
2. If I scroll to another position and then `btn.dispatchEvent(evt)`, within the event listener: 
   * **`pageY`**
       * Chrome: reflects the event's `clientY` value and never considers `scrollY` (initial or current), i.e. `123` ❌
       * WebKit: reflects the event's `clientY` value and never considers `scrollY` (initial or current), i.e. `123` ❌
       * Gecko: reflects the event's `clientY` value plus the current `scrollY` value ❌
         If the event listener scrolls (see commented-out code), `pageY` reflects the event's `clientY` value plus the newly changed `scrollY` value, i.e. `223` ❌
       * jsdom (prior to this PR): returns `0` ❌
       * jsdom (with this PR): reflects where the synthetic click happened relative to the document origin (i.e. the `scrollY` value at the time the event was created, plus `evt.clientY` which defaults to `0` absent a layout implementation) ✅
   * **`offsetY`**
       This is a weird one IMO. In order to get a reasonable value back, the event needs a reasonable `clientX/Y`. But even then browsers are inconsistent, and since jsdom doesn't implement `offsetTop` and friends, we can't return the same value as any browser. Additionally the browser `offsetX/Y`s are inconsistent with their own `pageX/Y`s, so something has to give (i.e. `pageX/Y` should be equal to `offsetX/Y` plus the target's `offsetLeft/offsetTop`, and that is not the case).
       * Chrome: reflects `evt.clientY + window.scrollY - btn.offsetTop`, i.e. `123 + ??? - 500` 🤷
         If the event listener scrolls (see commented-out code), `offsetY` changes accordingly; it's cached on the first access
       * WebKit: reflects (as of event init) `evt.clientY - btn.offsetTop`, i.e. `123 - 500` 🤷
       * Gecko: reflects (as of event init) `evt.clientY + window.scrollY - btn.offsetTop`, i.e. `123 + ??? - 500` 🤷
         If the event listener scrolls (see commented-out code), `offsetY` changes accordingly; it's recomputed each time
       * jsdom (prior to this PR): returns `0` ... at least it's consistent with scenarios 3 and 4 🤷
       * jsdom (with this PR): returns `0` ... at least it's consistent with scenarios 3 and 4 🤷
3. If I scroll to another position and then `btn.click()`, within the event listener:
   * **`pageY`**
      * Chrome: reflects the current `scrollY` value ❌
        If the event listener scrolls (see commented-out code), `pageY` reflects the value before the change ❌
      * WebKit: returns `0` ❌
      * Gecko: returns `0` ❌
      * jsdom (prior to this PR): returns `0` ❌
      * jsdom (with this PR): reflects where the synthetic click happened relative to the document origin (i.e. the `scrollY` value at the time the event was created, plus `evt.clientY` which defaults to `0` absent a layout implementation) ✅
   * **`offsetY`**
       I don't know that the spec spells out that it should be zero, but it's a reasonable default and all implementations agree
       * Chrome: returns `0` ✅
       * WebKit: returns `0` ✅
       * Gecko: returns `0` ✅
       * jsdom (prior to this PR): returns `0` ✅
       * jsdom (with this PR): returns `0` ✅
4. If I scroll to another position and then actually click, within the event listener:
   * **`pageY`**
      * Chrome: reflects where the click happened relative to the document origin (equivalent to `evt.clientY` plus the `scrollY` value at the time the event was created) ✅
      * WebKit: reflects where the click happened relative to the document origin (equivalent to `evt.clientY` plus the `scrollY` value at the time the event was created) ✅
      * Gecko: reflects where the click happened relative to the document origin (equivalent to `evt.clientY` plus the `scrollY` value at the time the event was created) ✅
      * jsdom (prior to this PR): n/a, can't actually click
      * jsdom (with this PR): n/a, can't actually click
   * **`offsetY`**
       * Chrome: returns relative offset from the target padding edge ✅
       * WebKit: returns relative offset from the target padding edge ✅
       * Gecko: returns relative offset from the target padding edge ✅
       * jsdom (prior to this PR): n/a, can't actually click
       * jsdom (with this PR): n/a, can't actually click


[^1]: Scrolling in a browser should be done by actually scrolling, or by doing something like `window.scroll(...)`. Setting `scrollX/Y` won't actually cause it to be used in calculations, nor will it actually scroll. In jsdom, OTOH, `scroll` and friends are not implemented, and thus "scrolling" must be done by setting `scrollX/Y`

Reply:
I should point out though, several of the ❌ could be thought of as ✅ if we’re not concerned about any scrolling (user or programmatic) that happens between event creation and retrieval of coordinates during dispatch. It probably never actually matters in the wild, except perhaps in weird synthetic dispatch tests like the ones I wrote 😆. 

Just trying to be thorough and ensure we don’t deviate from the spec unless 1. there’s a compelling reason to, and 2. major implementations are already in agreement (in which case the spec should change)

Reply:
Thanks @jsnajdr! I think I'm generally in agreement now, just a couple notes:
1. I think it's still a good idea to follow the spec and use `scrollX/Y` in the calculation, even if scrolling isn't fully supported yet. This way we don't need to revisit it when scrolling is implemented; at most some tests would just need to change to call `scroll/scrollBy` instead of setting those properties.
2. I'm going to expand the tests a little more around `offsetX/Y`, specifically I want to make sure we're well covered both in terms of 1. what we do implement spec-wise and 2. being clear around where we deviate due to not implementing layout support (e.g. either find and mark some existing WPTs as failing, or add some `dont-upstream` tests)

Reply:
> I think it's still a good idea to follow the spec and use `scrollX/Y` in the calculation, even if scrolling isn't fully supported yet.

I agree: returning `pageX: clientX + window.scrollX` and `offsetX: clientX`, both values look solid and shouldn't lead to any suprises :+1: